File: leet-code/romanToInt.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
  const map = {
    I: 1,
    V: 5,
    X: 10,
    L: 50,
    C: 100,
    D: 500,
    M: 1000
  }
  let result = 0
  let prev = map[s[0]]
  for (let i = 1; i < s.length; i++) {
    let currv = map[s[i]]
    if (prev < currv) {
      result -= prev
    } else {
      result += prev
    }
    prev = currv
  }
  result += prev
  return result
}

File: leet-code/romanToInt.go
package main

func romanToInt(s string) int {
	romanMap := make(map[byte]int)
	romanMap['I'] = 1
	romanMap['V'] = 5
	romanMap['X'] = 10
	romanMap['L'] = 50
	romanMap['C'] = 100
	romanMap['D'] = 500
	romanMap['M'] = 1000

	prev := romanMap[s[0]]
	result := 0
	for i := 1; i < len(s); i++ {
		currv := romanMap[s[i]]
		if prev < currv {
			result -= prev
		} else {
			result += prev
		}
		prev = currv
	}
	result += prev
	return result
}
